src/main/surfrdUtilsMod.F90

```diff
@@ -143,6 +145,205 @@ subroutine convert_cft_to_pft( begg, endg, cftsize, wt_cft )
+             wt_lunit(g,istcrop) = 0._r8
+             wt_cft(g,:) = 0._r8
+             wt_cft(g,nc3crop) = 1._r8  ! to pass error check below
+          else
+             do n = 1, n_dom_soil_patches
+                m = max_indices(n)
+                if (m >= cft_lb) then
+                   wt_cft(g,m) = wt_cft(g,m) / wt_dom_cft_sum
+                end if
+             end do
+          end if
+          ! Set non-dominant pft and cft weights to 0 for cases where dominant
+          ! pfts or cfts were found
+          do m = 0, maxveg
+             if (.not. any(max_indices == m)) then
+                if (m <= natpft_ub .and. wt_dom_nat_sum > 0._r8) then
+                   wt_nat_patch(g,m) = 0._r8
+                else if (m >= cft_lb .and. wt_dom_cft_sum > 0._r8) then
+                   wt_cft(g,m) = 0._r8
+                end if
+                ! Set to .F. regardless of whether dominant pfts and cfts were
+                ! found
+                pftcon%is_pft_known_to_model(m) = .false.
+             end if
+          end do
+
+       end do
+
+       write(iulog,*) 'is_pft_known...', pftcon%is_pft_known_to_model  ! slevis diag
+
+       ! Error checks
+       call check_sums_equal_1(wt_lunit, begg, 'wt_lunit', subname)
+       call check_sums_equal_1(wt_cft, begg, 'wt_cft', subname)
+       call check_sums_equal_1(wt_nat_patch, begg, 'wt_nat_patch', subname)
+
+       deallocate(max_indices)
+    end if
+
+  end subroutine collapse_all_pfts
+
+  !-----------------------------------------------------------------------
+  subroutine collapse_crop_var(crop_var, cft_size, begg, endg)
+    !
+    ! DESCRIPTION
+    ! After collapsing pft & cft weights in collapse_all_pfts, now ensure that
+    ! crop-related variables are consistent with the new crop weights (wt_cft).
+    !
+    ! Crop-related variables (locally named crop_var in this subroutine):
+    !
+    ! !USES:
+    use clm_varpar, only: cft_lb, cft_ub
+    use pftconMod, only: pftcon
+    !
+    ! !ARGUMENTS:
+    ! Use begg and endg rather than 'bounds', because bounds may not be
+    ! available yet when this is called
+    integer, intent(in) :: begg  ! Beginning grid cell index
+    integer, intent(in) :: endg  ! Ending grid cell index
+    integer, intent(in) :: cft_size  ! CFT size
+
+    ! Crop variable dimensioned [g, cft_lb:cft_lb+cft_size-1] modified in-place
+    real(r8), intent(inout) :: crop_var(begg:, cft_lb:)
+
+    ! !LOCAL VARIABLES:
+    integer :: g  ! gridcell index
+    integer :: m  ! cft index
+
+    character(len=*), parameter :: subname = 'collapse_crop_var'
+    !-----------------------------------------------------------------------
+
+    if (cft_size > 0) then  ! The opposite applies only if use_fates
+
+       SHR_ASSERT_ALL((ubound(crop_var) == (/endg, cft_lb+cft_size-1/)), errMsg(sourcefile, __LINE__))
+
+       do g = begg, endg
+          do m = cft_lb, cft_ub
+             if (pftcon%is_pft_known_to_model(m) == .false.) then
```

Comment:
Just started gnu standard testing and found that == should have been .eqv.

Reply:
or, better:

if (.not. pftcon%is_pft_known_to_model(m))

Reply:
Good point :-) Thanks!

Reply:
gnu standard testing PASSed by the way